crates/hdk_derive/src/unit_enum.rs

```diff
@@ -74,8 +77,11 @@ pub fn derive(input: TokenStream) -> TokenStream {
 
     // Forward any attributes that are meant for the unit enum.
     let unit_attrs: proc_macro2::TokenStream = match forward {
-        Some(syn::Meta::List(syn::MetaList { nested, .. })) => {
-            nested.iter().map(|a| quote::quote! {#[#a]}).collect()
+        Some(syn::Meta::List(syn::MetaList { tokens, .. })) => {
+            match Punctuated::<NestedMeta, Token![,]>::parse_terminated.parse(tokens.into()) {
+                Ok(nested) => nested.iter().map(|a| quote::quote! {#[#a]}).collect(),
+                Err(e) => return e.to_compile_error().into(),
+            }
```

Comment:
MetaList now holds a more generic token stream instead of nested metas. So the token stream needs to be parsed.